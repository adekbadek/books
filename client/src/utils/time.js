// @flow

import type moment from 'moment'
import type { Range } from 'utils/types'

export const DATE_FORMAT = 'DD MMM YYYY'

// TODO: refactor?

export const getRangesForDate = (date: moment, ranges: Array<Range>) => {
  let rangeNames = []
  ranges.map(range => {
    const is = range.start && range.end && date.isBetween(range.start, range.end, '', '[]')
    if (is) {
      rangeNames.push(range.name)
    }
    return is
  })
  return rangeNames
}
