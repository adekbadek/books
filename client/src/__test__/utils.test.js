import moment from 'moment'

import { isDateInRange } from '../utils/time.js'

it('isDateInRange', () => {
  expect(
    isDateInRange(moment('2017-05-09'), [{start: '2017-05-01', end: '2017-05-19'}])
  ).toBeTruthy()
  expect(
    isDateInRange(moment('2017-04-09'), [{start: '2017-05-01', end: '2017-05-19'}])
  ).not.toBeTruthy()
  expect(
    isDateInRange(moment('2016-05-09'), [{start: '2017-05-01', end: '2017-05-19'}])
  ).not.toBeTruthy()
})
