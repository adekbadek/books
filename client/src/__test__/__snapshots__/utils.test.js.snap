// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getCellStyles 1`] = `
Object {
  "backgroundColor": "rgb(30, 157, 0)",
}
`;

exports[`getCellStyles 2`] = `
Object {
  "backgroundColor": "hsla(108.5, 15%, 30.8%, 0.5)",
}
`;
